hil-test/tests/i2s_async.rs

```diff
@@ -0,0 +1,171 @@
+        peripherals.I2S0,
+        Standard::Philips,
+        DataFormat::Data16Channel16,
+        16000.Hz(),
+        dma_channel.configure_for_async(false, DmaPriority::Priority0),
+        tx_descriptors,
+        rx_descriptors,
+        &clocks,
+    );
+
+    let i2s_tx = i2s
+        .i2s_tx
+        .with_bclk(unsafe { io.pins.gpio0.clone_unchecked() })
+        .with_ws(unsafe { io.pins.gpio1.clone_unchecked() })
+        .with_dout(unsafe { io.pins.gpio2.clone_unchecked() })
+        .build();
+
+    let i2s_rx = i2s
+        .i2s_rx
+        .with_bclk(io.pins.gpio0)
+        .with_ws(io.pins.gpio1)
+        .with_din(io.pins.gpio3)
+        .build();
+
+    // enable loopback testing
+    unsafe {
+        let i2s = esp_hal::peripherals::I2S0::steal();
+        i2s.tx_conf().modify(|_, w| w.sig_loopback().set_bit());
+
+        i2s.rx_conf().modify(|_, w| w.rx_slave_mod().set_bit());
+
+        i2s.tx_conf().modify(|_, w| w.tx_update().clear_bit());
+        i2s.tx_conf().modify(|_, w| w.tx_update().set_bit());
+
+        i2s.rx_conf().modify(|_, w| w.rx_update().clear_bit());
+        i2s.rx_conf().modify(|_, w| w.rx_update().set_bit());
+    }
+
+    let mut iteration = 0;
+    let mut failed = false;
+    let mut check_i: u8 = 0;
+    let mut i = 0;
+    for b in tx_buffer.iter_mut() {
+        *b = i;
+        i = (i + ADD) % CUT_OFF;
+    }
+
+    let mut rcv = [0u8; 2000];
+
+    let mut rx_transfer = i2s_rx.read_dma_circular_async(rx_buffer).unwrap();
+    let tx_transfer = i2s_tx.write_dma_circular_async(tx_buffer).unwrap();
+
+    spawner.must_spawn(writer(i, tx_transfer));
+
+    'outer: loop {
+        let len = rx_transfer.pop(&mut rcv).await.unwrap();
+        for &b in &rcv[..len] {
+            if b != check_i {
+                failed = true;
+                break 'outer;
+            }
+            check_i = (check_i + ADD) % CUT_OFF;
+        }
+        iteration += 1;
+
+        if iteration > 30 {
+            break;
+        }
+    }
+
+    assert!(!failed);
+}
+
+#[embassy_executor::task]
+async fn writer(
+    i: u8,
+    mut transfer: I2sWriteDmaTransferAsync<
+        'static,
+        esp_hal::peripherals::I2S0,
+        DmaChannel0,
+        &'static mut [u8; 2000],
+    >,
+) {
+    let mut i = i;
+    loop {
+        transfer
+            .push_with(|buffer| {
+                for b in buffer.iter_mut() {
+                    *b = i;
+                    i = (i + ADD) % CUT_OFF;
+                }
+                buffer.len()
+            })
+            .await
+            .unwrap();
+    }
+}
+
+#[cfg(test)]
+#[embedded_test::tests]
+mod tests {
+    #[init]
+    fn init() {}
+
+    #[test]
+    fn run_test() {
+        unsafe fn __make_static<T>(t: &mut T) -> &'static mut T {
+            ::core::mem::transmute(t)
+        }
+
+        let mut executor = esp_hal_embassy::Executor::new();
+        let executor = unsafe { __make_static(&mut executor) };
+        executor.run(|spawner| {
```

Comment:
If we're doing all this to get the spawner, I think we can just use https://docs.embassy.dev/embassy-executor/git/cortex-m/struct.Spawner.html#method.for_current_executor to get the spawner?

Reply:
Ah I was looking for something like that but didn't find it 👍 will try